### PR TITLE
Partial term counts: don't affect other objects when preventing double counting.

### PR DESCRIPTION
**Pull request by peterwilsoncc**
Trac ticket: https://core.trac.wordpress.org/ticket/51630

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/taxonomy.php [hunk 1]
```diff
@@ -3283,25 +3283,38 @@ function wp_defer_term_counting( $defer = null ) {
...
+ *       - setting per object maybe, say `'post:10', $new_setting = null`.
+ *       - does it default to the current global post? Probably not.
+ *      *********************************** REMOVE THIS WILSON ***********
+ *
  * @private
  * @since 5.6.0
  *
- * @param bool $new_setting The new setting for preventing term counts.
+ * @param int    $object_id   The ID of the object.
+ * @param string $object_type Optional. Type of object: post, comment, term, user or site. Default post.
```
Comment (peterwilsoncc) [inline src/wp-includes/taxonomy.php:3301]:
```suggestion
 * @param string $object_type Optional. Type of object primitive: post, comment, term, user or site. Default post.
```

File: src/wp-includes/taxonomy.php [hunk 2]
```diff
@@ -2576,10 +2576,10 @@ function wp_set_object_terms( $object_id, $terms, $taxonomy, $append = false ) {
 
 	if ( array_filter( $object_types, 'post_type_exists' ) !== $object_types ) {
 		// This taxonomy applies to non-posts, count changes now.
-		$do_recount = ! _wp_prevent_term_counting();
+		$do_recount = ! _wp_prevent_term_counting( $object_id, reset( $object_types ) );
```
Comment (peterwilsoncc) [inline src/wp-includes/taxonomy.php:2579]: It'd be good here to make sure it's using the correct primitive.

**Review by peterwilsoncc — COMMENTED** [review]
notes to self...

Could be a problem if a taxonomy is registered for two primitive object types, eg `register_taxonomy( 'pwcc', [ 'post', 'user' ], $args )`, but it's a little out of warranty as there is no way of knowing if the relationship refers to post 1, or user 1.

**Comment by Chouby** [comment]
> Could be a problem if a taxonomy is registered for two primitive object types, eg `register_taxonomy( 'pwcc', [ 'post', 'user' ], $args )`, but it's a little out of warranty as there is no way of knowing if the relationship refers to post 1, or user 1.

Indeed, we can't do that for the exact reason that you exposed. In Polylang I had this need (define a taxonomy 'language' for posts and terms). Due to the fact that we cannot tell if the relationship refers to posts or terms or anything else, I needed to register 2 taxonomies, one for posts, one for terms.

**Pull request closed without merging**